### ISSUE TITLE
[a11y-defect-2]: Focus must be set to H2 when the Edit Mailing Address page loads

### ISSUE THREAD

**Issue by 1Copenut** [department-of-veterans-affairs/va.gov-team#104884]
### Point of contact

@1copenut

### Severity level

2, Serious. Should be fixed in 1-2 sprints post-launch.

### Details

The [Edit mailing address](https://github.com/department-of-veterans-affairs/vets-website/pull/34875) feature should set focus on the H2 when the edit view is loaded.

---
![Image](https://github.com/user-attachments/assets/d42d9012-cb3c-46ff-9507-e7ecdb053044)

### Reproduction steps

1. Log into Staging with a test account
2. Load the Your VA Letters and Documents view
3. Click the `Edit mailing address` link
4. Verify the focus is not on the H2 when the component loads
5. If you're using a screen reader, verify the "Edit mailing address" heading level two is not read aloud

### Proposed solution or next steps

We'll need to set focus on the H2 using a typical React solution with a `tabindex="-1"` and a `useEffect()` hook to set focus as the component renders.

### References, articles, or WCAG support

1. [Understanding SC 1.3.2: Meaningful Sequence (Level A)](https://w3.org/TR/WCAG22#meaningful-sequence)

### Type of issue

- [ ] Axe-core
- [x] Screenreader | assistive tech & device support
- [ ] Keyboard
- [ ] Zoom
- [ ] Color, typography, & visual elements
- [ ] Components and pattern standards
- [ ] Content organization
- [ ] Focus
- [ ] Headings
- [ ] Content style guide
- [ ] Link and button labels
- [ ] Semantic HTML
- [ ] Markup and metadata
- [ ] Non-text content and media
- [ ] Overall user experience
- [ ] Cognitive | user flows & navigating
- [ ] URL standards
- [ ] Something else

**Comment by salihaghaffar** [comment]
This is a known defect in staging, we have to fix it before it goes into staging review.

**Comment by dysbo** [comment]
✅ confirmed this is setting focus correctly with user +54
